File: 3212_Count_Submatrices_With_Equal_Frequency_of_X_and_Y/solution.c
int numberOfSubmatrices(char** g, int r, int* cs) {
    int c = cs[0];
    int x[c], y[c], ans = 0;
    for (int j = 0; j < c; j++) x[j] = y[j] = 0;
    for (int i = 0; i < r; i++) {
        int a = 0, b = 0;
        for (int j = 0; j < c; j++) {
            a += (g[i][j] == 'X');
            b += (g[i][j] == 'Y');
            x[j] += a;
            y[j] += b;
            if (x[j] > 0 && x[j] == y[j]) ans++;
        }
    }
    return ans;
}

File: 3212_Count_Submatrices_With_Equal_Frequency_of_X_and_Y/solution.py
class Solution:
    def numberOfSubmatrices(self, g: List[List[str]]) -> int:
        r, c = len(g), len(g[0])
        x = [0] * c
        y = [0] * c
        ans = 0
        for i in range(r):
            a = b = 0
            for j in range(c):
                a += g[i][j] == "X"
                b += g[i][j] == "Y"
                x[j] += a
                y[j] += b
                ans += x[j] > 0 and x[j] == y[j]
        return ans